[BUG]: compilation failure with fallthrough attribute

# Description
Compilation failure with [[fallthrough]]; not supported by all compiler version

1. Adding -Wimplicit-fallthrough=0
2. Fix version requiremetn for [[fallthrough]]

# Actual behaviour
Some old gcc version can't parse [[fallthrough]]
```
libcfg.c: In function ‘cfg_parse_line’:
libcfg.c:834:7: error: expected expression before ‘[’ token
  834 |       [[fallthrough]];
      |
```
# Extra informations or links
https://en.cppreference.com/w/c/language/attributes/fallthrough

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -2,7 +2,7 @@ SUBDIRS = .
 
 SOURCE_FILES = libcfg.c
 
-AM_CPPFLAGS = -I$(top_srcdir) -Wall -Werror -Wshadow -Wduplicated-cond -Wunused-parameter -fstack-protector
+AM_CPPFLAGS = -I$(top_srcdir) -Wall -Werror -Wshadow -Wduplicated-cond -Wunused-parameter -fstack-protector -Wimplicit-fallthrough=0
 
 if ENABLE_COVERAGE
 AM_CPPFLAGS += --coverage

--- a/src/libcfg.c
+++ b/src/libcfg.c
@@ -830,7 +830,7 @@ static cfg_parse_return_t cfg_parse_line(char *line, const size_t len,
       return CFG_PARSE_PASS;
     case CFG_PARSE_ARRAY_NEWLINE:
       *newline = ' ';
-#if __STDC_VERSION__ > 201112L
+#if __STDC_VERSION__ > 201710L
       [[fallthrough]];
 #endif
     case CFG_PARSE_CLEAN:
@@ -1436,7 +1436,7 @@ int cfg_read_file(cfg_t *cfg, const char *fname, const int prior) {
         case CFG_PARSE_ERROR:
           sprintf(msg, "%zu", nline);
           cfg_msg(cfg, "invalid configuration entry at line", msg);
-#if __STDC_VERSION__ > 201112L
+#if __STDC_VERSION__ > 201710L
           [[fallthrough]];
 #endif
         case CFG_PARSE_PASS:

